sql/core/src/main/scala/org/apache/spark/sql/execution/Window.scala

```diff
@@ -937,6 +945,12 @@ private[execution] object AggregateProcessor {
       imperatives.toArray,
       trackPartitionSize)
   }
+
+  // SPARK-14244: When binding
+  def windowPartitionSizeAttribute(expressions: Array[Expression]): AttributeReference = {
+    val aggs = expressions.flatMap(_.collectFirst { case f: SizeBasedWindowFunction => f })
+    aggs.headOption.map(_.n).getOrElse(SizeBasedWindowFunction.n)
```

Comment:
If there's no `SizeBasedWindowFunction`, actually we don't need the window partition size attribute, and invoking this function can be viewed as a no-op.